src/datapoints-graph/charts/charts.component.ts

```diff
@@ -224,14 +409,119 @@ export class ChartsComponent implements OnChanges, OnInit, OnDestroy {
     });
   }
 
+  private updatedTooltipFormatter(tooltipParams, params) {
```

Comment:
we have getTooltipFormatter in echarts options service and updatedTooltipFormatter in charts component. that is kind of confusing why we do have both. to make it more reasonable, both should go to service and should have names that makes obvious why we have two formatters